src/shared/mount-util.c

```diff
@@ -1053,32 +1053,70 @@ int make_mount_point(const char *path) {
+                * can just do it under its own regular UID. Note that in that case the container's runtime uidmap
+                * (i.e. the one the container payload processes run in) will leave this UID unmapped, i.e. if we
+                * accidentally leave files owned by host root in the already uidmapped tree around they'll show up
+                * as owned by 'nobody', which is safe. (Of course, we shouldn't leave such inodes around, but always
+                * chown() them to the container's own UID range, but it's good to have a safety net, in case we
+                * forget it.) */
+                if (idmapping == REMOUNT_IDMAPPING_HOST_ROOT) {
+                        if (strextendf(&line,
+                                       UID_FMT " " UID_FMT " " UID_FMT "\n",
+                                       UID_MAPPED_ROOT, 0, 1) < 0)
+                                return log_oom_debug();
+                }
+        }
+
+        if (idmapping == REMOUNT_IDMAPPING_HOST_OWNER) {
+                /* Remap the owner of the bind mounted directory to the root user within the container. This
+                 * way every file written by root within the container to bind-mounted directoryr will
+                 * be owned by the original user.
+                 * This requires 3 entries in uid_map:
+                 * - From host ROOT_UID to host OWNER_UID - 1, mapped starting from uid_shift.
+                 * - OWNER_UID mapped to uid_shift (root UID in container).
+                 * - From OWNER_UID + 1, mapped starting at the first free UID in uid_range.
+                 * However, there are 2 edge cases:
+                 * - If OWNER_UID is 0, then we proceed with identity mapping.
+                 * - If OWNER_UID is uid_range, then we will map all UIDs from 0 to uid_range -1 starting
+                 *      at uid_shift + 1, then map OWNER_UID to uid_shift.
+                 */
+                if (owner == 0) {
+                        if (asprintf(&line, UID_FMT " " UID_FMT " " UID_FMT "\n", 0, uid_shift, uid_range) < 0)
+                                return log_oom_debug();
+                } else if (owner == uid_range) {
+                        if (asprintf(&line,
+                                     UID_FMT " " UID_FMT " " UID_FMT "\n"
+                                     UID_FMT " " UID_FMT " " UID_FMT "\n",
+                                     0, uid_shift + 1, uid_range - 1,
+                                     owner, uid_shift, 1) < 0)
+                                return log_oom_debug();
+                } else {
+                        if (asprintf(&line,
+                                     UID_FMT " " UID_FMT " " UID_FMT "\n"
+                                     UID_FMT " " UID_FMT " " UID_FMT "\n"
+                                     UID_FMT " " UID_FMT " " UID_FMT "\n",
+                                     0, uid_shift + 1, owner,
```

Comment:
hmm, I am not sure I grok this: host uid 0 is mapped to uid 1 in the bind mount for the container? why that?